test/elements.js

```diff
@@ -104,6 +104,53 @@ test('adjacent text nodes', function (t) {
   t.end()
 })
 
+test('space in only-child text nodes', function (t) {
+  t.plan(1)
+  var result = bel`
+    <span>
+      surrounding
+      newlines
+    </span>
+  `
+  t.equal(result.outerHTML, '<span>surrounding\n      newlines</span>', 'should remove extra space')
```

Comment:
It's not so much about performance in this case, as it is about expectations.

This is what people write:
```html
<span>
  surrounding
  newlines
</span>
```

This is what people see rendered on the DOM:
```html
<span>surrounding newlines</span>
```

Now if people inspect the string content:
```js
// this matches what they see on the screen, and is probably wha
//  they expect the content to look like
var str = 'surrounding newlines'

// this is the actual representation - internals of how strings were parsed leak
var str = 'surrounding\n      newlines'
```

Tho it also goes worth noting strings can't be minified, so for complete applications the random empty spaces tend to add up.